### ISSUE TITLE
why ego_cls_loss keep increasing？ 

### ISSUE THREAD

**Issue by Zhaoguanhua** [jchengai/planTF#21]
In tensorboard,the curve of loss looks normal, it is decreasing.
![loss](https://github.com/jchengai/planTF/assets/33119925/d6517fa4-1f74-4d37-9edf-c5604f2d154c)

The simulation result of open loop and close loop(160W cache)  is close to numbers provided in the paper。
![指标](https://github.com/jchengai/planTF/assets/33119925/4c1f50bb-0994-4e26-b8a2-560615385dfb)
The visualization in nuboard also looks normal：
![start-left-turn2](https://github.com/jchengai/planTF/assets/33119925/e9519943-d1b8-47aa-8db4-8981ca452004)

everything looks like perfect，but when I check the loss，i found **the curve of the ego_cls_loss is always increasing** 
![objectives_loss](https://github.com/jchengai/planTF/assets/33119925/8f1fae56-308d-412a-b523-06ff0450b248)
(by the way,We have included the loss calculation in the objective module)，the ego_reg_loss and agent_reg_loss(predict loss) also looks normal，The trend during both the training and validation is same。

the ego_cls_loss is used to get the best trajectory id in six trajectorys from trajectory_decoder。
'trajectory, probability = self.trajectory_decoder(x[:, 0, :])'
the trajectory_decoder will generate six tajectorys for ego and each trajectory corresponds to a probability value。
the gt of ego_cls_loss is
```
ade = torch.norm(trajectory[..., :2] - ego_target[:, None, :, :2], dim=-1)
best_mode = torch.argmin(ade.sum(-1), dim=-1)
```
ego_cls_loss is computed by probability and best_mode,
`ego_cls_loss = F.cross_entropy(probability, best_mode.detach())`
so Isn't the normal ego_cls_loss curve supposed to decrease？why is it increasing during my train?

**Comment by jchengai** [comment]
Hi @Zhaoguanhua, I think this is pretty common in multi-modal prediction tasks. You may check https://github.com/ZikangZhou/HiVT/issues/15.
